assignment/tool/src/webapp/vm/assignment/chef_assignments_student_preview_submission.vm

```diff
@@ -113,6 +113,14 @@
 
     <p></p>
     <form id="addSubmissionForm" name="addSubmissionForm" action="#toolForm("AssignmentAction")" method="post" onsubmit="return true;">
+    	## We are allowing the instructor to submit as long as assignment is not non-electronic and not deleted
+		#if (!$submissionType != 4 && !$!deleted.equalsIgnoreCase("true") && $!allowAddAssignment && $!allowSubmitByInstructor && $!student) 
```

Comment:
All this logic in the template can be confusing, so unless your actually using each variable to display different pieces would it wouldn't be easier to inject a var in the context so that:
`#if (!$submissionType != 4 && !$!deleted.equalsIgnoreCase("true") && $!allowAddAssignment && $!allowSubmitByInstructor && $!student)`

becomes
```
#if(instructorCanSubmit)
...
#end
```

Reply:
I copied/pasted this from https://github.com/sakaiproject/sakai/blob/master/assignment/tool/src/webapp/vm/assignment/chef_assignments_student_view_submission.vm#L536

There was a bug with "preview" not retaining the state of `submit_on_behalf_of`, causing Sakai to redirect and make the submission count as the instructor's submission.